implementations/rust/ockam/ockam/src/entity/profile/profile_helpers.rs

```diff
@@ -0,0 +1,144 @@
+// FIXME: Give me better name, pls
+
+use crate::entity::profile::event_identifier::EventIdentifier;
+use crate::entity::profile::profile_change::{ProfileChange, ProfileKeyPurpose, ProfileKeyType};
+use crate::entity::profile::profile_change_type::ProfileChangeType::{CreateKey, RotateKey};
+use crate::entity::profile::proof::{ProfileChangeProof, Signature, SignatureType};
+use crate::entity::{Profile, ProfileChangeEvent};
+use crate::OckamError;
+use ockam_vault_core::secret::Secret;
+
+impl Profile {
+    pub(crate) fn get_last_event_id(&self) -> Result<EventIdentifier, ockam_core::Error> {
+        if let Some(e) = self.change_history.as_ref().last() {
+            Ok(e.identifier().clone())
+        } else {
+            Err(OckamError::InvalidInternalState.into())
+        }
+    }
+
+    pub(crate) fn find_key_change_in_event<'a>(
+        event: &'a ProfileChangeEvent,
+        label: &str,
```

Comment:
Should we have the ability to search by event id?
Might be necessary because there might be an event with duplicate labels, type and purpose.

Reply:
we haven't decided yet what is unique for specific key and what can be duplicated, that's why this was done based solely on my own vision

Reply:
Got it. Thoughts on just letting the caller search the events themselves?

Reply:
I'm not sure it's helpful, we can add it anytime